plugins/json/json.cxx

```diff
@@ -0,0 +1,392 @@
+			}
+		}
+	}
+
+    /** Write to json a pointer data
+	 *
+	 * \param json_data A json data to which we write a pointer data
+	 * \param reference A reference to a tuple datatype
+	 * \param logger PDI's logger instance
+	 */
+	void write_pointer_to_json(nlohmann::json& json_data, Ref_r reference, Logger& logger)
+	{
+		Ref dereferenced_ref = reference.dereference();
+		if (!dereferenced_ref) throw Value_error{"Can't dereference with read permissions"};
+		choose_type_and_dump_to_json(json_data, dereferenced_ref, logger);
+	}
+
+	/** Call different json dump function accordint to the datatype
+	 *
+	 * \param json_data A json data to which we write various types of data
+	 * \param reference A reference to a datatype
+	 * \param logger PDI's logger instance
+	 */
+	void choose_type_and_dump_to_json(nlohmann::json& json_data, Ref_r reference, Logger& logger)
+	{
+		if (const auto&& scalar_type = dynamic_pointer_cast<const Scalar_datatype>(reference.type())) { // a scalar type
+			write_scalar_to_json(json_data, reference, logger);
+		} else if (const auto&& array_type = dynamic_pointer_cast<const Array_datatype>(reference.type())) { // an array type
+			write_array_to_json(json_data, reference, logger);
+		} else if (const auto&& record_type = dynamic_pointer_cast<const Record_datatype>(reference.type())) { // a record type
+			write_record_to_json(json_data, reference, logger);
+		} else if (const auto&& pointer_type = dynamic_pointer_cast<const Tuple_datatype>(reference.type())) {
+			write_tuple_to_json(json_data, reference, logger);
+		} else if (const auto&& pointer_type = dynamic_pointer_cast<const Pointer_datatype>(reference.type())) { // a pointer type
+			write_pointer_to_json(json_data, reference, logger);
+		} else {
+			throw Type_error{"Unknown datatype passed to json"};
+		}
+	}
+
+	/** Write the variable to a JSON file
+	 *
+	 * \param data_name the variable name shared from PDI
+	 * \param indent_size reference The reference with read permission to this variable
+	 */
+	void write_data(const string& data_name, Ref_r reference)
+	{
+		Logger& logger = context().logger();
+
+		for (const auto& [condition, fpath]: m_data_to_path_map[data_name]) {
+			if (!condition.to_long(context())) {
+				logger.debug("Condition for {} isn't verified !", data_name);
+				continue;
+			}
+
+			const string filepath = fpath.to_string(context());
+
+			logger.debug("Writing data of {} for {}", data_name, filepath);
+			if (!reference) {
+				logger.error("Reading permissions were not granted for {}", data_name);
+			}
+
+			nlohmann::json json_data;
+			choose_type_and_dump_to_json(json_data[data_name], reference, logger);
+
+			path fp(filepath);
+			fstream json_file(fp, ios::in | ios::out | ios::ate);
+			if (!json_file.is_open()) {
+				// Case 1: File doesn't exist
+				// Create the file
+				// Write the json data between brackets
+				json_file.open(fp, ios::out);
+				json_file << "[\n" << json_data.dump(4) << "]";
```

Comment:
I'm not sure the square brackets are mandatory. It doesn't have to be a list.
I would remove the square brackets.
```suggestion
				json_file << json_data.dump(4);
```